kim_convergence/ase/equilibration.py

```
@@ -0,0 +1,258 @@
+        if extractors:
+            self.extractors.update(extractors)
+
+        self._extractors = []
+        for property_name in self.property_names:
+            # Validate that the requested property has an extractor
+            if property_name not in self.extractors:
+                available = ", ".join(sorted(self.extractors.keys()))
+                raise CRError(
+                    f"No extractor available for property: {property_name}. "
+                    f"Available properties: {available}"
+                )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)